engine/src/test/java/net/jqwik/api/EMailsTests.java

```diff
@@ -219,48 +220,70 @@ void domainsWithOneAndMorePartsAreGenerated(@ForAll("emails") String email){
 	@Group
 	class ShrinkingTests {
 
-		@Property
+		@Property(tries = 20)
+		void defaultShrinking(@ForAll Random random){
+			Arbitrary<String> emails = Arbitraries.emails();
+			assertAllValuesAreShrunkTo("A@a", emails, random);
+		}
+
+		@Property(tries = 20)
 		void domainShrinking(@ForAll Random random){
-			Arbitrary<String> emails = Arbitraries.emails().filter(v -> domainShrinkingFilter(v));
-			assertAllValuesAreShrunkTo("a@a", emails, random);
+			Arbitrary<String> emails = Arbitraries.emails();
+			Falsifier<String> falsifier = falsifyDomain();
+			String value = shrinkToMinimal(emails, random, falsifier);
+			assertThat(value).isEqualTo("A@a");
 		}
 
-		@Property
+		@Property(tries = 20)
 		void domainShrinkingWithTLD(@ForAll Random random){
-			Arbitrary<String> emails = Arbitraries.emails().filter(v -> domainShrinkingWithTLDFilter(v));
-			assertAllValuesAreShrunkTo("a@a.aa", emails, random);
+			Arbitrary<String> emails = Arbitraries.emails();
+			Falsifier<String> falsifier = falsifyDomainWithTLD();
+			String value = shrinkToMinimal(emails, random, falsifier);
+			assertThat(value).isEqualTo("A@a.aa");
 		}
```

Comment:
I don't know why this is not working. Do you have an idea?
`Expecting:
     <"A@a.a.aa">
    to be equal to:
     <"A@a.aa">
    but was not.`

Reply:
This shrinks to the correct value _most of the time_. Recursive shrinking is sometimes not fully stable because all shrinking depends on the initially created failing sample.

You could loosen the assertions a bit e.g. 
- Number of dots <= 3 (that's what I observed) and
- domain parts can only be "a" except for the TLD which must be "aa"

Getting a generator to _always_ shrink as expected is sometimes not possible and sometimes not worth it.

Reply:
Problem: Number of dots <= 3 does not work. I tried it with 10000 tries and got 4 dots. I think I only check the second point.